os.kieserver.launch/added/kieserver-setup.sh

```diff
@@ -31,25 +34,86 @@ function filterKieJmsFile() {
         fi
-        if [ "x${DB_JNDI}" != "x" ]; then
-            sed -i "s,org.quartz.dataSource.managedDS.jndiURL=,org.quartz.dataSource.managedDS.jndiURL=${DB_JNDI}," ${quartzPropFile}
+
+        if [ "x${DEFAULT_JNDI}" != "x" ]; then
             quartzManagedJndiSet="true"
         fi
         if [ "x${QUARTZ_JNDI}" != "x" ]; then
-            sed -i "s,org.quartz.dataSource.notManagedDS.jndiURL=,org.quartz.dataSource.notManagedDS.jndiURL=${QUARTZ_JNDI}," ${quartzPropFile}
             quartzNotManagedJndiSet="true"
         fi
+
+        if [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.MySQL"* ]]; then
+            quartzDriverDelegateSet="true"
+            DB_TYPE="MYSQL"
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.PostgreSQL"* ]]; then
+            QUARTZ_DELEGATE_CLASS="org.quartz.impl.jdbcjobstore.PostgreSQLDelegate"
+            quartzDriverDelegateSet="true"
+            DB_TYPE="POSTGRESQL"
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == *"Oracle"* ]]; then
+            log_warning "For Oracle 10G use the org.jbpm.persistence.jpa.hibernate.DisabledFollowOnLockOracle10gDialect hibernate dialect to avoid frequent Hibernate warning HHH000444 message in the logs."
+            QUARTZ_DELEGATE_CLASS="org.quartz.impl.jdbcjobstore.oracle.OracleDelegate"
+            # Oracle classes needs to be accessible by the quartz, in this case oracle driver or module needs to be added on jboss-deployment-structure.xml file
+            if [ "x${QUARTZ_DRIVER_MODULE}" != "x" ]; then
+                sed -i "s|<\!--EXTRA_DEPENDENCY-->|<module name=\"${QUARTZ_DRIVER_MODULE}\"/>\n      <\!--EXTRA_DEPENDENCY-->|" $JBOSS_HOME/standalone/deployments/kie-server.war/WEB-INF/jboss-deployment-structure.xml
+            else
+                log_warning "QUARTZ_DRIVER_MODULE env not set. Quartz might not work properly."
+            fi
+            quartzDriverDelegateSet="true"
+            DB_TYPE="ORACLE"
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.SQLServer"* ]]; then
+            QUARTZ_DELEGATE_CLASS="org.quartz.impl.jdbcjobstore.MSSQLDelegate"
+            quartzDriverDelegateSet="true"
+            DB_TYPE="SQLSERVER"
+
+            if [ "x${QUARTZ_DATABASE}" != "x" ]; then
+                local databaseName="${QUARTZ_DATABASE}"
+            fi
+
+            if [ "x${QUARTZ_XA_CONNECTION_PROPERTY_URL}" != "x" ]; then
+                local databaseName="${QUARTZ_XA_CONNECTION_PROPERTY_URL#*=}"
+            fi
+
+            if [ "x${QUARTZ_URL}" != "x" ]; then
+                local databaseName="${QUARTZ_URL#*=}"
+            fi
+
+            sed -i "s|USE \[enter_db_name_here\]|USE ${databaseName}|" $JBOSS_HOME/bin/quartz_tables_sqlserver.sql
+
+        elif [[ "${KIE_SERVER_PERSISTENCE_DIALECT}" == "org.hibernate.dialect.DB2"* ]]; then
+            if [ "x${KIE_SERVER_PERSISTENCE_SCHEMA}" != "x" ]; then
```

Comment:
I don't think it is strictly DB2 specific  (even I have only seen it used there).

Reply:
Yeah, Good point, I do remember to use it a long time ago with oracle.
I'll move this change next to the place where the kie.persistence.ds and dialect is configured.